pageserver/src/object_repository.rs

```diff
@@ -480,16 +554,16 @@ impl Timeline for ObjectTimeline {
     fn advance_last_record_lsn(&self, lsn: Lsn) {
         // Can't move backwards.
         let old = self.last_record_lsn.fetch_max(lsn);
-        assert!(old <= lsn);
-
-        // Also advance last_valid_lsn
-        let old = self.last_valid_lsn.advance(lsn);
-        // Can't move backwards.
-        if lsn < old {
-            warn!(
-                "attempted to move last record LSN backwards (was {}, new {})",
-                old, lsn
-            );
+        if old <= lsn {
```

Comment:
When does the `old > lsn' case happen?

Reply:
When we start replaying WAL from the beginning of segment